Enable GF Embedded test suite - part 3 (GF Embedded Maven plugin)

OndroMih opened the pull request:
Enables another part of GF Embedded tests for the maven plugin.

From my side, this is the last part of PRs that enable GF Embedded tests, I'm not going to work on this any longer There are still some disabled tests which might be worth enabling and fixing in the future.

OndroMih committed:
Enabled some of GF Embedded Maven plugin tests
Until websocket tests, which I only tried to refactor.
Signed-off-by:Ondro Mihalyi <mihalyi@omnifish.ee>

OndroMih committed:
Enabled almost all of GF Embedded Maven plugin tests
Only tests which fail and there's no simple fix are disabled.
Signed-off-by:Ondro Mihalyi <mihalyi@omnifish.ee>

OndroMih committed:
Merge eclipse/master into ondromih-gf-embedded-maven-plugin-tests

Conflicts:
	appserver/tests/embedded/pom.xml

Signed-off-by:Ondro Mihalyi <mihalyi@omnifish.ee>

OndroMih committed:
Upgraded GF Embedded maven plugin tests to JUnit 5
Signed-off-by:Ondro Mihalyi <mihalyi@omnifish.ee>

OndroMih commented:
The tests failed on some timeouts. Please, rerun the build, @dmatej , @avpinchuk , @arjantijms .

OndroMih committed:
Added enabled GF Embedded tests to the default test suites
Signed-off-by:Ondro Mihalyi <mihalyi@omnifish.ee>

OndroMih commented:
I forgot to add the GF Embedded test suite to the default test suites. I added it with the latest commit in this PR.

OndroMih committed:
Revert formatting done by my IDE
Signed-off-by:Ondro Mihalyi <mihalyi@omnifish.ee>

avpinchuk commented:
@OndroMih, local build failed too.

OndroMih committed:
Revemo GF Embedded tests from the fastest profile
Signed-off-by:Ondro Mihalyi <mihalyi@omnifish.ee>

OndroMih commented:
It was failing because the "fastest" profile sets "maven.skip.tests=true" to disable all tests. I removed the tests from the fastest profile.

OndroMih commented:
Again some tests failed on timeouts, please restart the build.

avpinchuk commented:
One moment, please ;)

Three local builds, one by one, passed.

dmatej commented on appserver/tests/pom.xml:
```diff
@@ -132,6 +133,7 @@
             <modules>
                 <!-- Builds compile dependencies for Ant tests executed by runtest.sh/gftest.sh -->
                 <module>appserv-tests</module>
+                <module>embedded</module>
```
I am not sure now, but fastest doesn't run any tests. appserv-tests are there just because they produce some artifacts required by additional ant tests executed in separate stage in jenkins.

OndroMih replied:
You are right. I didn't realize that at first, I removed embedded module from the `fastest` profile with a later commit.

dmatej reviewed (approved):
```
mvn clean install -pl :maven-plugin -amd
```
Passed, just some warmings:
```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for GlassFish Embedded Maven Plugin Tests 7.0.9-SNAPSHOT:
[INFO]
[INFO] GlassFish Embedded Maven Plugin Tests .............. SUCCESS [  1.824 s]
[INFO] Application Dispatcher Test (maven-plugin/applicationDispatcher) SUCCESS [  6.871 s]
[INFO] Directory Listing Test (maven-plugin/dirListing) ... SUCCESS [  2.156 s]
[INFO] Filter URI Mapping Test (maven-plugin/filterURIMapping) SUCCESS [  2.048 s]
[INFO] Get Request URI Test (maven-plugin/getRequestURI) .. SUCCESS [  0.529 s]
[INFO] Index Page Test (maven-plugin/index) ............... SUCCESS [  1.633 s]
[INFO] JMX Undeployment Event Test (maven-plugin/jmsUndeploymentTest) SUCCESS [  1.396 s]
[INFO] Query String Test (maven-plugin/queryString) ....... SUCCESS [  1.421 s]
[INFO] Session Lifecyle Test (maven-plugin/sessionDestroyed) SUCCESS [  1.533 s]
[INFO] Standalone WAR Test (maven-plugin/standalonewar) ... SUCCESS [  1.594 s]
[INFO] Test https and other plugin configurations (maven-plugin/secureWebApp) SUCCESS [  1.686 s]
[INFO] Test web app containing JSPs/JSTLs (maven-plugin/jsptest) SUCCESS [  2.248 s]
[INFO] Multiple application deployment test (maven-plugin/multipleApps) SUCCESS [  2.518 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  29.371 s
[INFO] Finished at: 2023-09-11T12:19:09+02:00
[INFO] ------------------------------------------------------------------------
[WARNING]
[WARNING] Plugin validation issues were detected in 3 plugin(s)
[WARNING]
[WARNING]  * org.apache.maven.plugins:maven-enforcer-plugin:3.1.0
[WARNING]  * org.glassfish.embedded:maven-embedded-glassfish-plugin:5.1
[WARNING]  * org.apache.maven.plugins:maven-war-plugin:3.3.2
[WARNING]
[WARNING] For more or less details, use 'maven.plugin.validation' property with one of the values (case insensitive): [BRIEF, DEFAULT, VERBOSE]
[WARNING]
```

Btw those start times are very good:
```
INFO: GlassFish  (commit: null) startup time: Embedded (36 ms), startup services (22 ms), total (58 ms)
```

dmatej commented on appserver/tests/embedded/maven-plugin/static-shell-init/pom.xml:
```diff
@@ -30,31 +31,43 @@
+                        <groupId>org.glassfish.embedded</groupId>
+                        <artifactId>maven-embedded-glassfish-plugin</artifactId>
+                        <dependencies>
+                            <dependency>
```
Formatting is a bit broken here

dmatej commented on appserver/tests/embedded/maven-plugin/standalonewar/src/main/java/test/ServletTest3.java:
```diff
@@ -17,31 +17,8 @@
 package test;
 
 import java.io.*;
-import java.net.*;
```
Could you yet replace those wildcard imports? Not sure how Netbeans, Eclipse can to that automatically.

OndroMih committed:
Disable running GF Embedded in tests if skipTests set
Signed-off-by:Ondro Mihalyi <mihalyi@omnifish.ee>

OndroMih committed:
Remove star imports
Signed-off-by:Ondro Mihalyi <mihalyi@omnifish.ee>

OndroMih committed:
Remove parameter 'goalPrefix' for GF Embedded Maven plugin.
This parameter was wrongly used. It's supposed to be used in the maven plugin itself to configure the prefix.
But this cannot be overriden in a maven project that uses the plugin.
Signed-off-by:Ondro Mihalyi <mihalyi@omnifish.ee>

OndroMih commented:
> ```
> mvn clean install -pl :maven-plugin -amd
> ```
> 
> Passed, just some warmings:
> 

@dmatej, I ran `mvn initialize -pl :maven-plugin -amd -Dmaven.plugin.validation=VERBOSE` with the following verbose output:

```
[WARNING] 
[WARNING]  * org.apache.maven.plugins:maven-enforcer-plugin:3.1.0
[WARNING]   Declared at location(s):
[WARNING]    * org.glassfish.main:nucleus-parent:7.0.9-SNAPSHOT (/home/ondro/workspaces/OmniFish/glassfish/nucleus/parent/pom.xml) @ line 1122
[WARNING]   Mojo EXTERNAL issue(s):
[WARNING]    * Mojo enforcer:enforce (org.apache.maven.plugins.enforcer.EnforceMojo)
[WARNING]      - Mojo implements `Contextualizable` interface from Plexus Container, which is EOL.
[WARNING] 
[WARNING]  * org.glassfish.embedded:maven-embedded-glassfish-plugin:5.1
[WARNING]   Plugin EXTERNAL issue(s):
[WARNING]    * Plugin is a Maven 2.x plugin, which will be not supported in Maven 4.x

```

it's not relared to changes in this PR. It's related to problems in the plugins:
* the enforcer plugin uses an obsolete API
* the GF Embedded plugin won't work in Maven 4.x as it's a Maven 2.x plugin
